### ISSUE TITLE
Exoplayer anr during release player.

### ISSUE THREAD

**Issue by JeffMony** [google/ExoPlayer#6078]
### [REQUIRED] Issue description
The exoplayer is doing release action, but anr is happening. This is very high probability.

### [REQUIRED] Reproduction steps
This is very high probability.

### [REQUIRED] Link to test content
Maybe some kind of execution is blocked by other things. 
Cause the UI thread to wait

### [REQUIRED] A full bug report captured from the device
"main" prio=5 tid=1 Waiting
  | group="main" sCount=1 dsCount=0 flags=1 obj=0x72fc9490 self=0xec637000
  | sysTid=23021 nice=-10 cgrp=default sched=0/0 handle=0xf386b4a4
  | state=S schedstat=( 25186301333 16640140943 57532 ) utm=1896 stm=622 core=3 HZ=100
  | stack=0xff2e9000-0xff2eb000 stackSize=8MB
  | held mutexes=
  at java.lang.Object.wait(Native method)
  - waiting on <0x03108b35> (a com.google.android.exoplayer2.PlayerMessage)
  at com.google.android.exoplayer2.PlayerMessage.k(PlayerMessage.java:282)
  - locked <0x03108b35> (a com.google.android.exoplayer2.PlayerMessage)
  at com.google.android.exoplayer2.SimpleExoPlayer.setVideoSurfaceInternal(SimpleExoPlayer.java:991)
  at com.google.android.exoplayer2.SimpleExoPlayer.setVideoSurfaceInternal(SimpleExoPlayer.java:1006)
  at com.google.android.exoplayer2.SimpleExoPlayer.setVideoSurface(SimpleExoPlayer.java:241)

"ExoPlayerImplInternal:Handler" prio=5 tid=95 Native
  | group="main" sCount=1 dsCount=0 flags=1 obj=0x1531c440 self=0xc18ea800
  | sysTid=26867 nice=-16 cgrp=default sched=0/0 handle=0xb7af9970
  | state=S schedstat=( 1055084445 502921515 7150 ) utm=74 stm=31 core=0 HZ=100
  | stack=0xb79f7000-0xb79f9000 stackSize=1038KB
  | held mutexes=
  kernel: __switch_to+0x94/0xa0
  kernel: futex_wait_queue_me+0xd8/0x130
  kernel: futex_wait+0xec/0x200
  kernel: do_futex+0xdc/0x8c0
  kernel: compat_SyS_futex+0xdc/0x168
  kernel: cpu_switch_to+0x26c/0x3a0
  native: #00 pc 00018dd4  /system/lib/libc.so (syscall+28)
  native: #01 pc 00046d11  /system/lib/libc.so (__pthread_cond_timedwait(pthread_cond_internal_t*, pthread_mutex_t*, bool, timespec const*)+102)
  native: #02 pc 000117f9  /system/lib/libstagefright_foundation.so (android::ALooper::awaitResponse(android::sp<android::AReplyToken> const&, android::sp<android::AMessage>*)+92)
  native: #03 pc 00012cf5  /system/lib/libstagefright_foundation.so (android::AMessage::postAndAwaitResponse(android::sp<android::AMessage>*)+120)
  native: #04 pc 000e4bbd  /system/lib/libstagefright.so (android::MediaCodec::PostAndAwaitResponse(android::sp<android::AMessage> const&, android::sp<android::AMessage>*)+20)
  native: #05 pc 000e7947  /system/lib/libstagefright.so (android::MediaCodec::flush()+82)
  native: #06 pc 000220d1  /system/lib/libmedia_jni.so (???)
  native: #07 pc 0000043b  /system/framework/arm/boot-framework.oat (Java_android_media_MediaCodec_native_1flush__+74)
  at android.media.MediaCodec.native_flush(Native method)
  at android.media.MediaCodec.flush(MediaCodec.java:2114)
  at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.flushCodec(MediaCodecRenderer.java:615)
  at com.google.android.exoplayer2.video.e.flushCodec(MediaCodecVideoRenderer.java:495)
  at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.onPositionReset(MediaCodecRenderer.java:466)
  at com.google.android.exoplayer2.video.e.onPositionReset(MediaCodecVideoRenderer.java:296)
  at com.google.android.exoplayer2.a.resetPosition(BaseRenderer.java:136)
  at com.google.android.exoplayer2.d.a(ExoPlayerImplInternal.java:712)
  at com.google.android.exoplayer2.d.a(ExoPlayerImplInternal.java:681)
  at com.google.android.exoplayer2.d.a(ExoPlayerImplInternal.java:642)
  at com.google.android.exoplayer2.d.a(ExoPlayerImplInternal.java:627)
  at com.google.android.exoplayer2.d.handleMessage(ExoPlayerImplInternal.java:294)
  at android.os.Handler.dispatchMessage(Handler.java:102)
  at android.os.Looper.loop(Looper.java:192)
  at android.os.HandlerThread.run(HandlerThread.java:65)

### [REQUIRED] Version of ExoPlayer being used
latest

### [REQUIRED] Device(s) and version(s) of Android being used
Vivo phone . android 8.0.1 android 6.x android 7.x

**Comment by andrewlewis** [comment]
This looks similar to #4352 except that the blocking call is `MediaCodec.flush()`. If you can post a full bug report captured via `adb bugreport` that may give clues to what is going on (and there's a chance this is a device-specific issue we can work around). Otherwise, I suggest marking this issue as a duplicate.

**Comment by ojw28** [comment]
Marking as a duplicate of #4352.
